yt/visualization/fits_image.py

```diff
@@ -1067,6 +1070,10 @@ class FITSProjection(FITSImageData):
     length_unit : string, optional
         the length units that the coordinates are written in. The default
         is to use the default length unit of the dataset.
+    moment : integer, optional
```

Comment:
I suggest we avoid mixing two different meaning of "optional" in the context of Python function signatures.
In terms of typing, `Optional` means "possibly `None`", but here it is not the case. (FTR this rather confusing name can be avoided in Python 3.9, but for the time being we _do_ use it in yt)
```suggestion
    moment : integer
```

Reply:
FWIW this is part of the `numpydoc` docstring standard: 

https://numpydoc.readthedocs.io/en/latest/format.html

and we've used it in plenty of other places in yt. I wasn't aware that we started discouraging this at any point.

Reply:
then carry on. I'm more familiar with `typing` than I am with numpy docstring style